verified_boot_main.md

```diff
@@ -9,25 +9,448 @@
+It is performed using asymetric cryptography. The private key of an asymmetric
+cryptography keypair is often called the `identity`. In this context, proving
+an identity is proving to be in possession of the private key.
+
+The simplest way an entity can prove it's identity is to encrypt a well known
+data using it's private key. If decrypting the data with a public
+key yields the same data, then it must have been encrypted using the
+corresponding private key.
+
+Verifying authenticity requires one to be in posession of a public key, that
+is trusted to correspond to the private key of the to bo authenticated entity.
+<!-- TODO? tell about how it is solved? certificate stores in UEFI / PKI? -->
+
+Authenticity itself does not guarantee the integrity of data.
+
+##### Non repudiation
+
+> A service that is used to provide assurance of the integrity and origin of data in such a way that the integrity and origin can be verified and validated by a third party as having originated from a specific entity in possession of the private key (i.e., the signatory).
+
+~ NIST FIPS 186-5 [^NIST_fips186-5]
+
+Non repudiation is a term used to describe a data, of which both the
+integrity and the authenticity of some entity responsible for it
+can be verified.
+Non repudiation is generally achieved using some form of digital
+signature.
+
+Digital Signature
+>  A cryptographic technique that utilizes asymmetric-keys to determine
+authenticity (i.e., users can verify that the message was signed with a private
+key corresponding to the specified public key), non-repudiation (a user cannot
+deny having sent a message) and integrity (that the message was not altered
+during transmission).
+
+~ NIST SP 800-63, Appendix A - Deifinitions and Abbreviations [^NIST_sp800-63]
+
+A basic digital signature is a digest of data, that has been encrypted using the
+private key of some entity.
+Verifying a signature requires:
+- The data in plaintext
+- The digital signature of the data
+- The public key corresponding to the private key of the signer
+- Knowledge of the hash function used to calculate the digest and the type of
+assymetric keys used by the signer
+
+<!-- TODO? digest is not necesarry, it just saves some time if the data is large, because asym. crypt. is slow -->
+
+The process consists of:
+- calculating the digest using the same hash function as used by the signer
+- decrypting the signature using signer's public key to receive the digest in
+  plaintext
+- comparing the two values
+
+The verification of the signature succeeds if both digests are exactly the same.
+If the verification succeeds then:
+- Integrity is verified. The received digest is the same as the one
+calculated from the datum. The data did not change
+<!-- TODO? only if the used hash function is safe? -->
+- Authenticity is verified. Only the one in possession of the corresponding
+private key could have encrypted the digest so that it can be decrypted using
+the public key
```

Comment:
This is valid, but slightly off topic. Let's keep it for now, but if the document becomes too long, this may be the first thing to be removed.